keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -161,146 +159,157 @@ consistent with the syntax supported by any of the common proxy vendors:
+  not allowed by [RFC-7230][rfc7230]0.
+* A request is a potential match for path `p_i` if `p_i` is the prefix of
+  the path. If there is more than one potential match, the longest
+  `p_i` will be the matching path.
+* If there is no matching path, then the `defaultBackend` for the host
+  will be used. If there is not match for the host, then the overall
+  `defaultBackend` for the Ingress will be selected.
 
-Regular expression subset (derived from [re2][re2-syntax] syntax page)
-
-| Expression | description             |
-|------------|-------------------------|
-| `.`        | any character           |
-| `[xyz]`    | character class         |
-| `[^xyz]`   | negated character class |
-| `x*`       | 0 or more x's           |
-| `x+`       | 1 or more x's           |
-| `xy`       | x followed by y         |
-| `x|y`      | x or y (prefer x)       |
-| `(abc)`    | grouping                |
-
-Maintaining a regular expression subset is likely not worth the complexity and
-is likely impossible across the [many implementations][regex-survey].
+[rfc7230]: https://tools.ietf.org/html/rfc7230#section-5.3.1
 
 ### API field naming
 
-These are straightforwarding one-to-one renames to for consistency.
+These are straightforwarding one-to-one renames for better semantic
+meaning.
 
-| v1beta1 field  | ga field              | rationale                   |
+| v1beta1 field  | v1                    | rationale                   |
 |----------------|-----------------------|-----------------------------|
 | `spec.backend` | `spec.defaultBackend` | Explicitly mentions default |
 
 ### Hostname wildcards
 
 Most platforms support wildcarding for hostnames, e.g. syntax such as
-`*.foo.com` matches names `app1.foo.com`, `app2.foo.com`. The current spec
-states that `spec.rules.host` must be a FQDN of a network host. 
-
-#### Proposal
-
-This proposal would be a limited support for adding a single wildcard `*` as the
-first host label.
-
-The `IngressRule.Host` comment would be changed to:
-
-> `Host` can be a "precise host" which is an fully-qualified domain name without
-> the terminating dot of a network host (e.g. "foo.bar.com") or a "wildcard
-> host" domain name prefixed with a single wildcard label ("*.foo.com").
-> Requests will be matched against the `Host` field in the following way:
-> 
-> If `Host` is precise, the request matches the rule if the http host header
-> is equal to `Host`.
-> 
-> If `Host` is a wildcard, then request matches the rule if the http host header
-> is to equal to the suffix (removing the first label) of the wildcard rule.
-> E.g. wildcard "*.foo.com" matches "bar.foo.com" because they share an equal
-> suffix "foo.com" but does NOT match "baz.bar.foo.com" because only the first
-> label is removed for a match.
+`*.foo.com` matches names `app1.foo.com`, `app2.foo.com`. The current
+spec states that `spec.rules.host` must be an exact FQDN match of a
+network host.
```

Comment:
This is correct but there is no validation avoiding the creation of `*.foo.com` in the host field

Reply:
Right, we would add the validation.